### PR TITLE
Add more tests to check mrow fallback when computing preferred logical widths

### PR DESCRIPTION
**Pull request by icesfont**
We currently check that certain invalid MathML markup falls back to being laid out as an `<mrow>` element. Add further tests to check that the preferred logical widths also fallback to being computed in the same way as an `<mrow>` element.

### CODE REVIEW

File: mathml/presentation-markup/mrow/mrow-fallback-002.html [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    done();
+  }
+</script>
+</head>
+<body>
+  <div id="log"></div>
+  <div id="container"></div>
+  <div id="invalidMultiscripts">
+    <div style="background: lightgreen; display: inline-block" class="invalid-mmultiscripts-container" data-description="first in-flow child is an <mprescripts>">
```
Comment (fred-wang) [inline mathml/presentation-markup/mrow/mrow-fallback-002.html:99]: I believe background: lightgreen is not necessary but maybe that's useful for developers debugging the tests. Up to you but please make it consistent with `wrapShrink()`. Also better to use a more neutral color such as gray as green often means PASS in WPT.

File: mathml/presentation-markup/mrow/mrow-fallback-002.html [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+          });
+          mspaceCount++;
+        }
+      }
+    });
+    Array.from(document.getElementById("invalidMultiscripts").
+               getElementsByClassName("invalid-mmultiscripts-container")).forEach(element => {
+      let reference = element.nextElementSibling;
+      let description = element.dataset.description;
+      console.log(reference, description)
```
Comment (fred-wang) [inline mathml/presentation-markup/mrow/mrow-fallback-002.html:85]: please remove console.log

File: mathml/presentation-markup/mrow/mrow-fallback-002.html [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+<script src="/mathml/support/mathml-fragments.js"></script>
+<style>
+  /* Revert style specified in the UA style sheet that changes box size. */ 
+  mfrac { padding-inline: 0; }
+</style>
+<script>
+  setup({ explicit_done: true });
+  window.addEventListener("load", () => { loadAllFonts().then(runTests); });
+
+  function isValid(tagName, mspaceCount) {
```
Comment (fred-wang) [inline mathml/presentation-markup/mrow/mrow-fallback-002.html:22]: I think this is duplicating a lot of code from `mrow-fallback-002.html`. Can you please factor out some code into a `mathml/presentation-markup/mrow/support/invalid-markup.js` file? I guess that would be `isValid()` and the code generating the invalid markup.

**Review by fred-wang — CHANGES_REQUESTED** [review]

File: mathml/presentation-markup/mrow/mrow-fallback-002.html (hunk 1 above)
Comment (icesfont) [reply mathml/presentation-markup/mrow/mrow-fallback-002.html]: Removed in `invalid-markup.js`

File: mathml/presentation-markup/mrow/mrow-fallback-002.html (hunk 2 above)
Comment (icesfont) [reply mathml/presentation-markup/mrow/mrow-fallback-002.html]: Removed

File: mathml/presentation-markup/mrow/mrow-fallback-002.html (hunk 3 above)
Comment (icesfont) [reply mathml/presentation-markup/mrow/mrow-fallback-002.html]: Done

File: mathml/presentation-markup/mrow/mrow-fallback-001.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    let invalidMarkup = generateInvalidMarkup();
+    container.appendChild(invalidMarkup);
+    const epsilon = 1;
+
+    for (let math of invalidMarkup.children) {
+      let element = math.firstElementChild;
+      let reference = element.nextElementSibling;
+      let description = element.dataset.description;
+      test(function() {
+        compareLayout(element, reference, epsilon);
```
Comment (fred-wang) [inline mathml/presentation-markup/mrow/mrow-fallback-001.html:34]: can you please also add a test for the baseline alignment here?

Probably

`assert_approx_equals(elementWrapper.getBoundingClientRect().top, referenceWrapper.getBoundingClientRect().top, epsilon);`

would be enough.

**Review by fred-wang — CHANGES_REQUESTED** [review]
Thanks. This LGTM, just with the baseline check added.

File: mathml/presentation-markup/mrow/mrow-fallback-001.html (hunk 4 above)
Comment (icesfont) [reply mathml/presentation-markup/mrow/mrow-fallback-001.html:34]: Added (without the wrapper)

**Review by fred-wang — APPROVED** [review]

**Pull request merged**